test/integration/testsuite/minishift.go

```diff
@@ -72,6 +74,45 @@ func (m *Minishift) profileShouldHaveState(profile string, expected string) erro
 	return nil
 }
 
+func (m *Minishift) ShouldHaveNoOfProcessors(noOfprocessor int) error {
+	cpuInfo, err := m.runner.CpuInfo()
+	if err != nil {
+		return err
+	}
+	cpuString := "processor\\s*:\\s*\\d"
+	re := regexp.MustCompile(cpuString)
+	listItems := re.FindAllString(cpuInfo, -1)
+	if len(listItems) != noOfprocessor {
+		return fmt.Errorf("The vm is running with %d no. of cpus. Expected: %d", len(listItems), noOfprocessor)
+	}
+	return nil
+}
+
+//This function is only applicable for validating disk size between 11-98(GB/G/gb/g)
+//applied through minishift disk-size flag
+func (m *Minishift) ShouldHaveDiskSize(diskSizeString string) error {
```

Comment:
Function still seems too complicated to me, something I could imagine is:

```
func shouldHaveDiskSize(expectedSize string) error {
	expected, _ := strconv.Atoi(expectedSize)

	diskInfo, err := DiskInfo()
	//some regexp to get "25.0" out of "Disk /dev/sda: 25.0 GB, ........"
	actual, err := strconv.ParseFloat("25.0", 64)
	
	if int(actual) != expected {
		return fmt.Errorf("The vm is running with disk size of %v. Expected: %v", actual, expected)
	}

	return nil
}
```

I am wondering how @LalatenduMohanty sees it for example.

Reply:
Or because exact values of disk vary from hypervisor to hypervisor, maybe interval would be the clearest solution:
```
func shouldHaveDiskSize(min string, max string) error {
    [...]
    if min < int(actual) < max {
        return nil
    }
    return fmt.Erorrf(...)
}
```

Reply:
Looks good and simple.

Reply:
Done